File: day1/windturbines/submissions/partially_accepted/charlotte_path.cpp
#include<bits/stdc++.h>

using namespace std;

signed main(){
    int n,m,q;
    cin >> n >> m >> q;;
    vector<array<long long,3>> edges;

    for (int i=0;i<m;++i){
        long long u,v,c;
        cin >> u >> v >> c;
        edges.push_back({u,v,c});
    }
    sort(edges.begin(),edges.end());

   vector<long long> pref(n,0);
   for(int i=1;i<n;++i) {
     pref[i] = pref[i-1]+edges[i-1][2];
   }

    for (int i=0;i<q;++i){
        int l,r;
        cin >> l >> r;
        cout << pref[n-1]-(pref[r]-pref[l])<< '\n';

}
}

File: day1/windturbines/submissions/partially_accepted/charlotte_path.py
n, m, q = map(int, input().split())
edges = []
for _ in range(m):
    u, v, c = map(int, input().split())
    edges.append((u, v, c))

edges.sort()
pref = [0 for _ in range(n)]
for i in range(1, n):
    pref[i] = pref[i - 1] + edges[i - 1][2]

for _ in range(q):
    l, r = map(int, input().split())
    print(pref[n - 1] - (pref[r] - pref[l]))
